src/Fantomas/TokenParser.fs

```diff
@@ -547,11 +547,43 @@ let private (|InterpStringEndOrPartToken|_|) token =
 let escapedCharacterRegex =
     System.Text.RegularExpressions.Regex("(\\\\(a|b|f|n|r|t|u|v|x|'|\\\"|\\\\))+")
 
-let rec private (|EndOfInterpolatedString|_|) tokens =
+let private splitAround<'a> (n: int) (xs: 'a list) : ('a list * ('a * 'a list) option) option =
+    let rec go (n: int) (heads: 'a list) (rest: 'a list) =
+        if n = 0 then
+            match rest with
+            | [] -> Some(List.rev heads, None)
+            | x :: rest -> Some(List.rev heads, Some(x, rest))
+        else
+            match rest with
+            | [] -> None
+            | x :: rest -> go (n - 1) (x :: heads) rest
+
+    go n [] xs
```

Comment:
Is there a suitable place for this function? It seems like it could be useful elsewhere. If not, I could just leave as is or make it an inner function of `EndOfInterpolatedString`.

Reply:
If it were made internal and put somewhere at the top level, it could be tested!

There's a nice simple property-based test this could be covered by (assuming there's any prior art for FsCheck in Fantomas, which I don't know):

```fsharp
[<Test>]
let ``my test`` () =
    let property (xs : int list) (n : int) : bool =
        let actual = List.splitAround n xs
        match actual with
        | None -> n > List.length xs
        | Some (init, r) ->
            n <= List.length xs &&
            match r with
            | None -> xs = init
            | Some (x, tail) ->
                xs = init @ (x :: tail)
    Check.QuickThrowOnFailure property
```

Reply:
A number of ways this property-based test could be improved, by the way. The most obvious to me: split into three tests, one for `n > List.length xs`, one for `n < List.length xs`, and one for `n = List.length xs`. For example:

```fsharp
open FsCheck
[<Test>]
let ``When n equals list length`` () =
    let property (xs : int list) : bool =
        let actual = List.splitAround (List.length xs) xs
        match actual with
        | None -> false
        | Some (init, None) -> init = xs
        | Some (init, Some _) -> false
    Check.QuickThrowOnFailure property

[<Test>]
let ``When n less than list length`` () =
    let property (xs : int list, n : int) : bool =
        let actual = List.splitAround (List.length xs) xs
        match actual with
        | None -> false
        | Some (init, None) -> init = xs
        | Some (init, Some _) -> false

    let gen =
        gen {
            let! xs = Gen.list<int> // or however you do this
            let len = List.length xs
            let! n = Gen.int |> Gen.filter (fun n -> n < len)
            return (xs, n)
        }

    property
    |> Prop.forAll (Arb.fromGen gen)
    |> Check.QuickThrowOnFailure
```

Reply:
> Is there a suitable place for this function?

How about https://github.com/fsprojects/fantomas/blob/4c3aa3d6c3d26a396f1781da614000b7ec90d406/src/Fantomas/Utils.fs.

Matching unit test file: https://github.com/fsprojects/fantomas/blob/4d73011bb2d0e96b4c15cac9262dfe8b0757e0e6/src/Fantomas.Tests/UtilsTests.fs

Reply:
Great, thanks for the suggestions.